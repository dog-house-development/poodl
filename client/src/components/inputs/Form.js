import React, { Component } from "react";
import PropTypes from "prop-types";
import Field from "./Field";
import Button from "./Button";

const propTypes = {
  onSubmit: PropTypes.func.isRequired,
  fields: PropTypes.array.isRequired,
  noValidate: PropTypes.bool,
  buttonLabel: PropTypes.string
};

const defaultProps = {
  buttonLabel: "Submit"
};

class Form extends Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.getFieldsMarkup = this.getFieldsMarkup.bind(this);
  }

  onComponentDidMount() {
    this.setState({ value: this.props.content });
  }

  handleChange(e) {
    e.preventDefault();

    this.props.onChange(e);
  }

  getFieldsMarkup() {
    return this.props.fields.map(field => {
      return (
        <Field
          key={field.id}
          id={field.id}
          type={field.type}
          name={field.name}
          label={field.label}
          content={field.content}
          placeholder={field.placeholder}
          onChange={field.onChange}
          error={field.error}
        />
      );
    });
  }

  render() {
    return (
      <form noValidate={this.props.noValidate} onSubmit={this.props.onSubmit}>
        {this.getFieldsMarkup()}
        <Button type="submit" content={this.props.buttonLabel} />
      </form>
    );
  }
}

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;
export default Form;
