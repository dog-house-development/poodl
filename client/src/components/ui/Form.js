import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import Field from './Field';
import Button from './Button';

const propTypes = {
    onSubmit: PropTypes.func.isRequired,
    fields: PropTypes.array.isRequired,
    formTitle: PropTypes.string,
    noValidate: PropTypes.bool,
    buttonLabel: PropTypes.string
};

const defaultProps = {
    buttonLabel: 'Submit'
};

class Form extends Component {
    getFieldsMarkup() {
        return _.map(this.props.fields, field => (
            <Field
                key={field.id}
                id={field.id}
                type={field.type}
                name={field.name}
                label={field.label}
                content={field.content}
                placeholder={field.placeholder}
                onChange={field.onChange}
                error={field.error}
            />
        ));
    }

    render() {
        return (
            <div className="panel">
                <h1 className="panel-title">{this.props.formTitle}</h1>
                <form noValidate={this.props.noValidate} onSubmit={this.props.onSubmit}>
                    {this.getFieldsMarkup()}
                    <Button type="submit" content={this.props.buttonLabel} formButton />
                </form>
            </div>
        );
    }
}

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;
export default Form;
