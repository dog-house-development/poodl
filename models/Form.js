const mongoose = require('mongoose');
const Schema = mongoose.Schema;
const ObjectId = mongoose.Schema.Types.ObjectId;

const formSchema = new Schema(
    {
        seniorCenterId: {
            type: ObjectId,
            required: true
        },
        fields: {
            type: String,
            required: [true, 'Fields are required']
        },
        name: {
            type: String,
            required: [true, 'Name is required']
        },
        description: {
            type: Date,
            required: false
        }
    },
    { timestamps: true }
);

module.exports = formSchema;
