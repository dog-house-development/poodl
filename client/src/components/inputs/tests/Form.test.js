import React from "react";
import ReactDOM from "react-dom";
import Form from "../Form";

it("renders without crashing", () => {
  const props = {
    fields: [{ id: "name", onChange: () => {} }],
    onSubmit: () => {}
  };
  const div = document.createElement("div");
  ReactDOM.render(<Form {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
