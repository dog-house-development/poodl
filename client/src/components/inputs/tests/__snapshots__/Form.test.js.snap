// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form tests render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    buttonLabel="Submit"
    fields={
      Array [
        Object {
          "error": Object {},
          "id": "email",
          "label": "Email:",
          "onChange": [Function],
          "placeholder": "Enter email...",
          "type": "email",
          "value": "",
        },
      ]
    }
    noValidate={false}
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Field
            error={Object {}}
            id="email"
            label="Email:"
            onChange={[Function]}
            placeholder="Enter email..."
            size="normal"
            type="email"
          />,
        ],
        <Button
          content="Submit"
          kind="primary"
          size="medium"
          type="submit"
        />,
      ],
      "noValidate": false,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "email",
        "nodeType": "class",
        "props": Object {
          "content": undefined,
          "error": Object {},
          "id": "email",
          "label": "Email:",
          "name": undefined,
          "onChange": [Function],
          "placeholder": "Enter email...",
          "size": "normal",
          "type": "email",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "content": "Submit",
          "kind": "primary",
          "size": "medium",
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Field
              error={Object {}}
              id="email"
              label="Email:"
              onChange={[Function]}
              placeholder="Enter email..."
              size="normal"
              type="email"
            />,
          ],
          <Button
            content="Submit"
            kind="primary"
            size="medium"
            type="submit"
          />,
        ],
        "noValidate": false,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "email",
          "nodeType": "class",
          "props": Object {
            "content": undefined,
            "error": Object {},
            "id": "email",
            "label": "Email:",
            "name": undefined,
            "onChange": [Function],
            "placeholder": "Enter email...",
            "size": "normal",
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "content": "Submit",
            "kind": "primary",
            "size": "medium",
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
